crates/analytics/src/auth_events/metrics/challenge_attempt_count.rs

```diff
@@ -0,0 +1,90 @@
+use api_models::analytics::{
+    auth_events::{AuthEventFlows, AuthEventMetricsBucketIdentifier},
+    Granularity, TimeRange,
+};
+use common_utils::errors::ReportSwitchExt;
+use error_stack::ResultExt;
+use time::PrimitiveDateTime;
+
+use super::AuthEventMetricRow;
+use crate::{
+    query::{Aggregate, GroupByClause, QueryBuilder, QueryFilter, ToSql, Window},
+    types::{AnalyticsCollection, AnalyticsDataSource, MetricsError, MetricsResult},
+};
+
+#[derive(Default)]
+pub(super) struct ChallengeAttemptCount;
+
+#[async_trait::async_trait]
+impl<T> super::AuthEventMetric<T> for ChallengeAttemptCount
+where
+    T: AnalyticsDataSource + super::AuthEventMetricAnalytics,
+    PrimitiveDateTime: ToSql<T>,
+    AnalyticsCollection: ToSql<T>,
+    Granularity: GroupByClause<T>,
+    Aggregate<&'static str>: ToSql<T>,
+    Window<&'static str>: ToSql<T>,
+{
+    async fn load_metrics(
+        &self,
+        merchant_id: &str,
+        _publishable_key: &str,
+        granularity: &Option<Granularity>,
+        time_range: &TimeRange,
+        pool: &T,
+    ) -> MetricsResult<Vec<(AuthEventMetricsBucketIdentifier, AuthEventMetricRow)>> {
+        let mut query_builder: QueryBuilder<T> =
+            QueryBuilder::new(AnalyticsCollection::ConnectorEvents);
```

Comment:
we are querying for connectorevents & deserializing to authevents? 😕

Reply:
AuthEventMetricRow would only have time_bucket and count, so we can retain this